test/end-to-end/core.sh

```diff
@@ -84,191 +85,192 @@ wait_for_registry
-[[ -n ${pusher_token} ]]
+os::cmd::expect_success_and_text "echo ${pusher_token}" '.+'
 
 echo "[INFO] Docker login as pusher to ${DOCKER_REGISTRY}"
-docker login -u e2e-user -p ${pusher_token} -e pusher@openshift.com ${DOCKER_REGISTRY}
+os::cmd::expect_success "docker login -u e2e-user -p ${pusher_token} -e pusher@openshift.com ${DOCKER_REGISTRY}"
 echo "[INFO] Docker login successful"
 
 # log back into docker as e2e-user again
-docker login -u e2e-user -p ${e2e_user_token} -e e2e-user@openshift.com ${DOCKER_REGISTRY}
+os::cmd::expect_success "docker login -u e2e-user -p ${e2e_user_token} -e e2e-user@openshift.com ${DOCKER_REGISTRY}"
 
 echo "[INFO] Back to 'default' project with 'admin' user..."
-oc project ${CLUSTER_ADMIN_CONTEXT}
-[ "$(oc whoami | grep 'system:admin')" ]
+os::cmd::expect_success "oc project ${CLUSTER_ADMIN_CONTEXT}"
+os::cmd::expect_success_and_text 'oc whoami' 'system:admin'
 
 # The build requires a dockercfg secret in the builder service account in order
 # to be able to push to the registry.  Make sure it exists first.
 echo "[INFO] Waiting for dockercfg secrets to be generated in project 'test' before building"
-wait_for_command "oc get -n test serviceaccount/builder -o yaml | grep dockercfg > /dev/null" $((60*TIME_SEC))
+os::cmd::try_until_text 'oc get -n test serviceaccount/builder -o yaml' 'dockercfg'
 
 # Process template and create
 echo "[INFO] Submitting application template json for processing..."
 STI_CONFIG_FILE="${ARTIFACT_DIR}/stiAppConfig.json"
 DOCKER_CONFIG_FILE="${ARTIFACT_DIR}/dockerAppConfig.json"
 CUSTOM_CONFIG_FILE="${ARTIFACT_DIR}/customAppConfig.json"
-oc process -n test -f examples/sample-app/application-template-stibuild.json > "${STI_CONFIG_FILE}"
-oc process -n docker -f examples/sample-app/application-template-dockerbuild.json > "${DOCKER_CONFIG_FILE}"
-oc process -n custom -f examples/sample-app/application-template-custombuild.json > "${CUSTOM_CONFIG_FILE}"
+os::cmd::expect_success "oc process -n test -f examples/sample-app/application-template-stibuild.json > ${STI_CONFIG_FILE}"
+os::cmd::expect_success "oc process -n docker -f examples/sample-app/application-template-dockerbuild.json > ${DOCKER_CONFIG_FILE}"
+os::cmd::expect_success "oc process -n custom -f examples/sample-app/application-template-custombuild.json > ${CUSTOM_CONFIG_FILE}"
 
 echo "[INFO] Back to 'test' context with 'e2e-user' user"
-oc login -u e2e-user
-oc project test
-oc whoami
+os::cmd::expect_success 'oc login -u e2e-user'
+os::cmd::expect_success 'oc project test'
+os::cmd::expect_success 'oc whoami'
 
 echo "[INFO] Running a CLI command in a container using the service account"
-oc policy add-role-to-user view -z default
+os::cmd::expect_success 'oc policy add-role-to-user view -z default'
 out=$(oc run cli-with-token --attach --env=POD_NAMESPACE=test --image=openshift/origin:${TAG} --restart=Never -- cli status --loglevel=4 2>&1)
-echo "${out}"
```

Comment:
@liggitt Since we run everything in a subshell, I don't see how we can get the pesky `out=$()` lines into a wrapper function. Is this something we can run a couple of times? I guess this is a second use-case for testing a command's output for multiple strings.